libs/sdk-ui-dashboard/src/model/_staging/dashboard/dashboardLayout.ts

```diff
@@ -0,0 +1,130 @@
+// (C) 2021 GoodData Corporation
+
+import {
+    DashboardWidget,
+    IDashboardLayout,
+    IDashboardLayoutItem,
+    ISettings,
+    isInsightWidget,
+    isKpiWidget,
+    WidgetType,
+} from "@gooddata/sdk-backend-spi";
+import { IInsight, insightRef, serializeObjRef } from "@gooddata/sdk-model";
+import { MeasurableWidgetContent, validateDashboardLayoutWidgetSize } from "@gooddata/sdk-ui-ext";
+import compact from "lodash/compact";
+import keyBy from "lodash/keyBy";
+import { InvariantError } from "ts-invariant";
+
+function extractContentFromWidget(
+    widget: DashboardWidget,
+    insightsById: Record<string, IInsight>,
+): { type: WidgetType; content?: MeasurableWidgetContent } {
+    if (isInsightWidget(widget)) {
+        const insightRef = widget.insight;
+        const insight = insightsById[serializeObjRef(insightRef)];
+
+        if (!insight) {
```

Comment:
Detail: the if is redundant. Just
```
return {
    type: "insight",
    content: insight,
};
```
should be enough, right?

Reply:
fixed